includes/WikiPage.php

```diff
@@ -1659,8 +1663,11 @@ public function doEditUpdates( Revision $revision, User $user, array $options =
 		}
 
 		# Update the links tables
-		$u = new LinksUpdate( $this->mTitle, $editInfo->output );
-		$u->doUpdate();
+		// SRE-109: tell LinksUpdate if this is a new page and defer it till the end of the request
+		$linksUpdate = new LinksUpdate( $this->mTitle, $editInfo->output );
+		$linksUpdate->setWasPageNewlyCreated( $options['created'] );
+
+		DeferredUpdates::addUpdate( $linksUpdate );
```

Comment:
Are you sure that subsequent operations don't rely on the result of this one?

Reply:
Yeah, if something was relying on that, chances are they'd have used the `LinksUpdateComplete` hook which will still work just a same.